polyfactory/value_generators/constrained_numbers.py

```diff
@@ -110,23 +111,37 @@ def passes_pydantic_multiple_validator(value: T, multiple_of: T) -> bool:
     """
     if multiple_of == 0:
         return True
-    mod = float(value) / float(multiple_of) % 1
-    return almost_equal_floats(mod, 0.0) or almost_equal_floats(mod, 1.0)
+    mod = value % multiple_of
+    return almost_equal_floats(float(mod), 0.0) or almost_equal_floats(float(abs(mod)), float(abs(multiple_of)))
 
 
-def get_increment(t_type: type[T]) -> T:
+def get_increment(value: T, t_type: type[T]) -> T:
```

Comment:
This would be a breaking change. Instead, we should have another `get_increment_v2` function and deprecate this one.